WordPress/Classes/Services/PostHelper.m

```diff
@@ -249,12 +249,12 @@ + (NSArray *)mergePosts:(NSArray <RemotePost *> *)remotePosts
 {
     NSMutableArray *posts = [NSMutableArray arrayWithCapacity:remotePosts.count];
     for (RemotePost *remotePost in remotePosts) {
-        AbstractPost *post;
-        NSUUID *foreignID = remotePost.foreignID;
-        if (foreignID != nil) {
-            post = [blog lookupPostWithForeignID:foreignID inContext:context];
-        } else {
-            post = [blog lookupPostWithID:remotePost.postID inContext:context];
+        AbstractPost *post = [blog lookupPostWithID:remotePost.postID inContext:context];
```

Comment:
I started testing by looking into this area. The first lookup that should happen is the `postID` lookup. The current version performs only a `foreignID` lookup. I can't come up with scenarios where this might be an issue, but it's better to keep the `postID` logic in place.